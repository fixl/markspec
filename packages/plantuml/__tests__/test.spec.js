const { toSvg, toSvgSync, check, checkSync, toUri, toUriSync } = require("..");

const diagram = `
@startuml
A-->B: request
B-->A: ack
@enduml`;

const erroneousDiagram = `@start`;

describe("plantuml", () => {
  it("renders a diagram", async () => {
    const syncResult = toSvgSync(diagram);
    const result = await toSvg(diagram);
    const syncUri = toUriSync(diagram);
    const uri = await toUri(diagram);
    expect(result).toMatchSnapshot();
    expect(result).toEqual(syncResult);
    expect(uri).toMatchSnapshot();
    expect(uri).toEqual(syncUri);
});
  it("renders an erroneous diagram by default", () => {
    const result = toSvgSync(erroneousDiagram);
    expect(result).toMatchSnapshot();
  });
  it("checks syntax", async () => {
    await check(diagram);
    checkSync(diagram);
    expect(() => {
      checkSync(erroneousDiagram);
    }).toThrow();
    await expect(check(erroneousDiagram)).rejects.toMatchSnapshot();
  });
});
