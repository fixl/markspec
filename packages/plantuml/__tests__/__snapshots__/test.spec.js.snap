// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plantuml checks syntax 1`] = `[Error: plantuml syntax error]`;

exports[`plantuml renders a diagram 1`] = `"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" standalone=\\"no\\"?><svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" contentScriptType=\\"application/ecmascript\\" contentStyleType=\\"text/css\\" height=\\"161px\\" preserveAspectRatio=\\"none\\" style=\\"width:109px;height:161px;\\" version=\\"1.1\\" viewBox=\\"0 0 109 161\\" width=\\"109px\\" zoomAndPan=\\"magnify\\"><defs><filter height=\\"300%\\" id=\\"fii913vv8i4cw\\" width=\\"300%\\" x=\\"-1\\" y=\\"-1\\"><feGaussianBlur result=\\"blurOut\\" stdDeviation=\\"2.0\\"/><feColorMatrix in=\\"blurOut\\" result=\\"blurOut2\\" type=\\"matrix\\" values=\\"0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 .4 0\\"/><feOffset dx=\\"4.0\\" dy=\\"4.0\\" in=\\"blurOut2\\" result=\\"blurOut3\\"/><feBlend in=\\"SourceGraphic\\" in2=\\"blurOut3\\" mode=\\"normal\\"/></filter></defs><g><line style=\\"stroke: #A80036; stroke-width: 1.0; stroke-dasharray: 5.0,5.0;\\" x1=\\"21\\" x2=\\"21\\" y1=\\"39.6094\\" y2=\\"120.3125\\"/><line style=\\"stroke: #A80036; stroke-width: 1.0; stroke-dasharray: 5.0,5.0;\\" x1=\\"88\\" x2=\\"88\\" y1=\\"39.6094\\" y2=\\"120.3125\\"/><rect fill=\\"#FEFECE\\" filter=\\"url(#fii913vv8i4cw)\\" height=\\"31.6094\\" style=\\"stroke: #A80036; stroke-width: 1.5;\\" width=\\"23\\" x=\\"8\\" y=\\"3\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"14\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"9\\" x=\\"15\\" y=\\"24.5332\\">A</text><rect fill=\\"#FEFECE\\" filter=\\"url(#fii913vv8i4cw)\\" height=\\"31.6094\\" style=\\"stroke: #A80036; stroke-width: 1.5;\\" width=\\"23\\" x=\\"8\\" y=\\"119.3125\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"14\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"9\\" x=\\"15\\" y=\\"140.8457\\">A</text><rect fill=\\"#FEFECE\\" filter=\\"url(#fii913vv8i4cw)\\" height=\\"31.6094\\" style=\\"stroke: #A80036; stroke-width: 1.5;\\" width=\\"23\\" x=\\"75\\" y=\\"3\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"14\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"9\\" x=\\"82\\" y=\\"24.5332\\">B</text><rect fill=\\"#FEFECE\\" filter=\\"url(#fii913vv8i4cw)\\" height=\\"31.6094\\" style=\\"stroke: #A80036; stroke-width: 1.5;\\" width=\\"23\\" x=\\"75\\" y=\\"119.3125\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"14\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"9\\" x=\\"82\\" y=\\"140.8457\\">B</text><polygon fill=\\"#A80036\\" points=\\"76.5,67.9609,86.5,71.9609,76.5,75.9609,80.5,71.9609\\" style=\\"stroke: #A80036; stroke-width: 1.0;\\"/><line style=\\"stroke: #A80036; stroke-width: 1.0; stroke-dasharray: 2.0,2.0;\\" x1=\\"21.5\\" x2=\\"82.5\\" y1=\\"71.9609\\" y2=\\"71.9609\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"13\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"43\\" x=\\"28.5\\" y=\\"67.1045\\">request</text><polygon fill=\\"#A80036\\" points=\\"32.5,98.3125,22.5,102.3125,32.5,106.3125,28.5,102.3125\\" style=\\"stroke: #A80036; stroke-width: 1.0;\\"/><line style=\\"stroke: #A80036; stroke-width: 1.0; stroke-dasharray: 2.0,2.0;\\" x1=\\"26.5\\" x2=\\"87.5\\" y1=\\"102.3125\\" y2=\\"102.3125\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"13\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"21\\" x=\\"38.5\\" y=\\"97.4561\\">ack</text></g></svg>"`;

exports[`plantuml renders a diagram 2`] = `"data:image/svg+xml,%3c%3fxml version='1.0' encoding='UTF-8' standalone='no'%3f%3e%3csvg xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' contentScriptType='application/ecmascript' contentStyleType='text/css' height='161px' preserveAspectRatio='none' style='width:109px%3bheight:161px%3b' version='1.1' viewBox='0 0 109 161' width='109px' zoomAndPan='magnify'%3e%3cdefs%3e%3cfilter height='300%25' id='fii913vv8i4cw' width='300%25' x='-1' y='-1'%3e%3cfeGaussianBlur result='blurOut' stdDeviation='2.0'/%3e%3cfeColorMatrix in='blurOut' result='blurOut2' type='matrix' values='0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 .4 0'/%3e%3cfeOffset dx='4.0' dy='4.0' in='blurOut2' result='blurOut3'/%3e%3cfeBlend in='SourceGraphic' in2='blurOut3' mode='normal'/%3e%3c/filter%3e%3c/defs%3e%3cg%3e%3cline style='stroke: %23A80036%3b stroke-width: 1.0%3b stroke-dasharray: 5.0%2c5.0%3b' x1='21' x2='21' y1='39.6094' y2='120.3125'/%3e%3cline style='stroke: %23A80036%3b stroke-width: 1.0%3b stroke-dasharray: 5.0%2c5.0%3b' x1='88' x2='88' y1='39.6094' y2='120.3125'/%3e%3crect fill='%23FEFECE' filter='url(%23fii913vv8i4cw)' height='31.6094' style='stroke: %23A80036%3b stroke-width: 1.5%3b' width='23' x='8' y='3'/%3e%3ctext fill='black' font-family='sans-serif' font-size='14' lengthAdjust='spacingAndGlyphs' textLength='9' x='15' y='24.5332'%3eA%3c/text%3e%3crect fill='%23FEFECE' filter='url(%23fii913vv8i4cw)' height='31.6094' style='stroke: %23A80036%3b stroke-width: 1.5%3b' width='23' x='8' y='119.3125'/%3e%3ctext fill='black' font-family='sans-serif' font-size='14' lengthAdjust='spacingAndGlyphs' textLength='9' x='15' y='140.8457'%3eA%3c/text%3e%3crect fill='%23FEFECE' filter='url(%23fii913vv8i4cw)' height='31.6094' style='stroke: %23A80036%3b stroke-width: 1.5%3b' width='23' x='75' y='3'/%3e%3ctext fill='black' font-family='sans-serif' font-size='14' lengthAdjust='spacingAndGlyphs' textLength='9' x='82' y='24.5332'%3eB%3c/text%3e%3crect fill='%23FEFECE' filter='url(%23fii913vv8i4cw)' height='31.6094' style='stroke: %23A80036%3b stroke-width: 1.5%3b' width='23' x='75' y='119.3125'/%3e%3ctext fill='black' font-family='sans-serif' font-size='14' lengthAdjust='spacingAndGlyphs' textLength='9' x='82' y='140.8457'%3eB%3c/text%3e%3cpolygon fill='%23A80036' points='76.5%2c67.9609%2c86.5%2c71.9609%2c76.5%2c75.9609%2c80.5%2c71.9609' style='stroke: %23A80036%3b stroke-width: 1.0%3b'/%3e%3cline style='stroke: %23A80036%3b stroke-width: 1.0%3b stroke-dasharray: 2.0%2c2.0%3b' x1='21.5' x2='82.5' y1='71.9609' y2='71.9609'/%3e%3ctext fill='black' font-family='sans-serif' font-size='13' lengthAdjust='spacingAndGlyphs' textLength='43' x='28.5' y='67.1045'%3erequest%3c/text%3e%3cpolygon fill='%23A80036' points='32.5%2c98.3125%2c22.5%2c102.3125%2c32.5%2c106.3125%2c28.5%2c102.3125' style='stroke: %23A80036%3b stroke-width: 1.0%3b'/%3e%3cline style='stroke: %23A80036%3b stroke-width: 1.0%3b stroke-dasharray: 2.0%2c2.0%3b' x1='26.5' x2='87.5' y1='102.3125' y2='102.3125'/%3e%3ctext fill='black' font-family='sans-serif' font-size='13' lengthAdjust='spacingAndGlyphs' textLength='21' x='38.5' y='97.4561'%3eack%3c/text%3e%3c/g%3e%3c/svg%3e"`;

exports[`plantuml renders an erroneous diagram by default 1`] = `"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" standalone=\\"no\\"?><svg xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\" contentScriptType=\\"application/ecmascript\\" contentStyleType=\\"text/css\\" height=\\"23px\\" preserveAspectRatio=\\"none\\" style=\\"width:146px;height:23px;background:#000000;\\" version=\\"1.1\\" viewBox=\\"0 0 146 23\\" width=\\"146px\\" zoomAndPan=\\"magnify\\"><defs/><g><text fill=\\"#33FF02\\" font-family=\\"sans-serif\\" font-size=\\"14\\" font-weight=\\"bold\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"139\\" x=\\"5\\" y=\\"19.5332\\">No @startuml found</text></g></svg>"`;
